src/simulation/simulation_store_common.jl

```diff
@@ -0,0 +1,180 @@
+# Keep these in sync with the Symbols in src/core/definitions.
+get_store_container_type(::AuxVarKey) = STORE_CONTAINER_AUX_VARIABLES
+get_store_container_type(::ConstraintKey) = STORE_CONTAINER_DUALS
+get_store_container_type(::ExpressionKey) = STORE_CONTAINER_EXPRESSIONS
+get_store_container_type(::ParameterKey) = STORE_CONTAINER_PARAMETERS
+get_store_container_type(::VariableKey) = STORE_CONTAINER_VARIABLES
+
+function write_model_dual_results!(
+    store::SimulationStore,
+    model::DecisionModel,
+    timestamp::Dates.DateTime,
+    exports,
+)
+    container = get_optimization_container(model)
+    model_name = get_name(model)
+    if exports !== nothing
+        exports_path = joinpath(exports[:exports_path], "duals")
+        mkpath(exports_path)
+    end
+
+    for (key, constraint) in get_duals(container)
+        write_result!(
+            store,
+            model_name,
+            STORE_CONTAINER_DUALS,
+            key,
+            timestamp,
+            constraint,
+            [encode_key(key)],  # TODO DT: is this what the columns should be?
```

Comment:
This old comment applies to a current problem. We need a rule here and a common way to do it.

Reply:
agreed, we should discuss how to make this consistent for one dimensional variables.